plugins/shared-state.ts

```diff
@@ -0,0 +1,27 @@
+import { provideSharedStates } from './shared-state/provider.ts'
+import { defineDynamicStateHolder } from './shared-state/dynamic-holder.ts'
+
+export default defineNuxtPlugin(async () => {
+  return {
+    provide: {
+      ...provideSharedStates({
+        static: {
+          tags: await useTagsFetch(),
+        },
+        dynamic: {
+          homepageProjects: defineDynamicStateHolder({
+            key: 'homepage-projects',
+            state: useFetch('/api/homepage-projects'),
+            ttl: 10 * 60,
```

Comment:
TTL'ing it on the server route would be enough

Reply:
Not really, no, this is client's own TTL, which may not match the TTL on the server.

Reply:
It may not, but this scenario is not nearly complicated enough to grant a need for 2 separate TTLs

Reply:
On the server this TTL is never used, it's purely clint-side, server has TTL for the route for 30 minutes after the first invocation. Here the client will consider data stale after 10 minutes and try to fetch again. I mean it can be changed to match the server, but again, there will be mismatch between when server first cached the route and when client got that data, but not that homepage projects is such a critical data to re-fetch frequently.

Reply:
You misunderstood. There’s no much need to store it client-side with a TTL at all, the payload size is negligible and the amount of visits to the homepage is small